### ISSUE TITLE
ACL field extractions not working properly

### ISSUE THREAD

**Issue by kennykentaur** [inspired/TA-cisco_ios#23]
**Splunk 7.2.0 and TA-cisco_ios 2.5.4 on single instance.**

When querying acl data it doesn't properly extract `src_ip, src_port, dest_ip, dest_port` 

Tried the field extraction for acl in `transforms.conf` against the supplied sample data at https://regexr.com and it's seems to get lost after processing the first statements up until src_ip.

**Comment by inspired** [comment]
Hi!
Could you send me some log samples?

**Comment by kennykentaur** [comment]
> 
> 
> Hi!
> Could you send me some log samples?

Sure, I'll send some tomorrow! However I think you can verify the issue with the sample data that you included with the app.

**Comment by kennykentaur** [comment]
Sent log samples by mail.

**Comment by inspired** [comment]
I have tested this in both rubular.com and regexr.com with your samples and I cannot reproduce what you are seeing. Are you sure you haven't made any local modifications to the regex?

![screenshot 2018-11-06 at 10 06 10](https://user-images.githubusercontent.com/52004/48053704-a34a1a80-e1ab-11e8-890c-39908f950ead.png)

The below regex matches just fine:

`(IPACCESSLOGP|IPACCESSLOGSP|IPACCESSLOGRP|IPACCESSLOGNP|ACCESSLOGP|ACCESSLOGSP|ACCESSLOGNP)(\s)?:(?:.+)list\s(?<rule>.+)\s(?<vendor_action>denied|permitted)\s(?<proto>\d+|tcp|udp|igmp|ipinip|gre|eigrp|ospf|nosip|pim|sctp)\s(?<src_ip>(?:(?:\d{1,3}\.){3}(?:\d{1,3}))|(?:(?:::)?(?:[\dA-Fa-f]{1,4}:{1,2}){1,7}(?:[\d\%A-Fa-z\.]+)?(?:::)?)|(?:::[\dA-Fa-f\.]{1,15})|(?:::))\(?(?<src_port>\d+)?\)?(\s\((?<src_interface>\S+) (?<src_mac>\S+)\))?\s->\s(?<dest_ip>(?:(?:\d{1,3}\.){3}(?:\d{1,3}))|(?:(?:::)?(?:[\dA-Fa-f]{1,4}:{1,2}){1,7}(?:[\d\%A-Fa-z\.]+)?(?:::)?)|(?:::[\dA-Fa-f\.]{1,15})|(?:::))(\(?(?<dest_port>\d+)?\))?(, (?<packets>\S+) packet(s)?)?(\s\[(?<correlation_tag>\S+)\])?`

**Comment by kennykentaur** [comment]
Hi,

Yes, I've tested the regex again on regexr.com and you're right. I must have messed something up the first time around.

I've done some more troubleshooting, but I'm none the wiser.
If I paste the REGEX as a `| rex <regexp>` it also works.

I dont have any local directory for the TA with any settings of my own, also confirmed by a btool.

[https://pastebin.com/akWafvXt](https://pastebin.com/akWafvXt)

Either I'm hitting a 7.2 issue or there something else touching the data for extractions

**Comment by kennykentaur** [comment]
I've tried the 2.5.4 TA-cisco_ios on a 6.6.3 instance and that is working like a charm. 
This is either a 7.2.0 or another TA interfering. 

I'm closing this issue.

**Comment by kennykentaur** [comment]
The latest commits [https://github.com/inspired/TA-cisco_ios/commit/21b45c090d3a9f02ec4fe1434bfd3878c29e8751](21b45c0) and 
[https://github.com/inspired/TA-cisco_ios/commit/9fcc25dbe14cfae908e99b08993b47ad8114e852](9fcc25d) fixed the issue I had with the `dest_ip `field missing from the ACL extractions.

However I still don't get any extraction for `src_ip `and `action`. 

Commenting out the following lines from `props.conf` got src_ip extraction working for the ACL's, however I haven't gotten around anything to address the extraction for `action`

```
FIELDALIAS-cisco_ios-neighbor = neighbor AS src_ip
FIELDALIAS-cisco_ios-cdp_remote_ip = cdp_remote_ip AS src_ip
```

**Comment by kulcsari** [comment]
Hi, 

I also removed the `port_status AS vendor_action` from the `FIELDALIAS-cisco_ios-various`. After this, the action status is shown. So far I don't test the side effects... 
This is with Splunk 7.2.2. I've already tested the original addon with Splunk v7.0.4 with no problem. I cannot imagine what changed.

István

**Comment by Wolvverine** [comment]
in Security ACL dashbord:
![image](https://user-images.githubusercontent.com/5873929/52491604-cb408400-2bc7-11e9-9e0c-2912faf588e6.png)

regexp is ok:
![image](https://user-images.githubusercontent.com/5873929/52491838-486bf900-2bc8-11e9-81b0-e7c413bf2f2f.png)

I removed the `port_status AS vendor_action` from the `FIELDALIAS-cisco_ios-various`. After this, the action status is shown.

And now the field "action"  in event view  is also shown.

**Comment by Wolvverine** [comment]
" A field can have multiple aliases, but a single alias can only apply to one field. For example, the field vendor_action can be aliased to action or message_type, but not both."

![image](https://user-images.githubusercontent.com/5873929/52496293-40fe1d00-2bd3-11e9-92f0-5a58440c445f.png)

**Comment by Wolvverine** [comment]
@inspired 
The problem with fields "dest_ip" not showing the data in  Cisco Network App -> security -> ACL is because the field name is "dest_ip" in Cisco Network App and "dest" in Cisco Network Add-on.
When in dashbord change "dest_ip" to "dest" - fields are shown correctly (but only in this dashboard).

**Comment by inspired** [comment]
Thanks for all the input. If anyone is willing to test the changes and make
a pull request, I will be happy to merge it.

Den fre. 8. feb. 2019, 22:08 skrev Michał Panasiewicz <
notifications@github.com>:

**Comment by Wolvverine** [comment]
I'm not sure that the problem is only in the acl dashboard. This could be wider fields/aliases incompatibility between the application and the add-on. 
Fields:
dest_ip,dest,...
src_ip,src,cdp_remote_ip,neighbor,.....
vendor_action, port_status,.....
src_port
dest_port

**Comment by inspired** [comment]
I have made the following changes to TA-cisco_ios as of b855881e8b9f7f8e1461b3d2751815b6223a5f5f
Dropped the following and extracted the aliases directly instead:
FIELDALIAS-cisco_ios-neighbor = neighbor AS src_ip
FIELDALIAS-cisco_ios-cdp_remote_ip = cdp_remote_ip AS src_ip
FIELDALIAS-cisco_ios-various = dropped port_status to vendor_action

No changes made to the app as of yet.

**Comment by Wolvverine** [comment]
Problem with extract src_ip - solved
remove:
FIELDALIAS-cisco_ios-nat = nat_inside_local_ip AS src_ip
or in props.conf add:
FIELDALIAS-cisco_ios-nat = nat_inside_local_port AS src_port, nat_inside_global_ip AS dest_ip, nat_inside_global_port AS dest_port, nat_outside_global_ip AS dest_translated_ip, nat_outside_global_port AS dest_translated_port, nat_outside_local_ip AS src_translated_ip, nat_outside_local_port AS src_translated_port

and remove/comment out:
FIELDALIAS-cisco_ios-neighbor = neighbor AS src_ip
FIELDALIAS-cisco_ios-cdp_remote_ip = cdp_remote_ip AS src_ip
FIELDALIAS-cisco_ios-src_ip = src_ip AS src

**Comment by Wolvverine** [comment]
all problems solved:

FIELDALIAS-cisco_ios-nat =  nat_outside_global_ip AS dest_translated_ip, nat_outside_global_port AS dest_translated_port, nat_outside_local_ip AS src_translated_ip, nat_outside_local_port AS src_translated_port
#FIELDALIAS-cisco_ios-cdp_remote_ip = cdp_remote_ip AS src_ip
#FIELDALIAS-cisco_ios-neighbor = neighbor AS src_ip
#FIELDALIAS-cisco_ios-src_ip = src_ip AS src
#FIELDALIAS-cisco_ios-dest_ip = dest_ip AS dest

**Comment by Wolvverine** [comment]
duplicate fields: "proto" and "protocol", leave only "proto":

#FIELDALIAS-cisco_ios-proto = proto AS protocol
#FIELDALIAS-cisco_ios-transport = proto AS transport

**Comment by inspired** [comment]
@Wolvverine are you sure you don't have a local/props.conf or a local/transforms.conf in your install as the following is already removed as it is not required because the fields are now named correctly in the extractions:

FIELDALIAS-cisco_ios-nat = nat_outside_global_ip AS dest_translated_ip, nat_outside_global_port AS dest_translated_port, nat_outside_local_ip AS src_translated_ip, nat_outside_local_port AS src_translated_port

For CIM compliance I am keeping the following and I cannot understand why it shouldn't work:
FIELDALIAS-cisco_ios-src_ip = src_ip AS src
FIELDALIAS-cisco_ios-dest_ip = dest_ip AS dest
FIELDALIAS-cisco_ios-transport = protocol AS transport

All the other stuff has been renamed in the extractions as of 4687cf6c95fb4a6abf745ebef8de074ec6168914

**Comment by inspired** [comment]
If you could try TA-cisco-ios 4c31cf6  and https://github.com/inspired/cisco_ios/commit/5398a05493e426c4a104f499bcb01c067752d88c together that would be great.

**Comment by kennykentaur** [comment]
I've tried the versions now with your latest commits and as far as I can see the extracts are fully working now, thank you!

The only thing I'm seeing now is the vendor_message lookup rendering without filling out event info([chars] rather then the data from the event), see screenshow. But that is a part of the cisco_ios app rather then the TA, so I'll raise the issue there instead.

![capture6](https://user-images.githubusercontent.com/34913130/52862017-6a1f3000-3134-11e9-9bc4-f5bd00d8951a.PNG)

**Comment by inspired** [comment]
This is expected as the vendor_message_info is the description of the
formatting of the event as expected from the device. It is not supposed to
be filling out any info. The actual info is in the message field or _raw
field of the event :-)

Glad to hear it's now fixed.

Den fre. 15. feb. 2019, 15:15 skrev kennykentaur <notifications@github.com>:

**Comment by inspired** [comment]
@kennykentaur I modified the overview page to show the latest value of the message_text field instead of the "template". Fixed in cisco_ios commit ae0b71cc32726a05d72f7e26eccddb933007032a

**Comment by inspired** [comment]
I am closing this as I believe the issues are now resolved. Please let me know if this is not the case.
